pipelines/batch/src/main/java/org/openmrs/analytics/FetchSearchPageFn.java

```diff
@@ -142,7 +147,8 @@ private static synchronized JdbcConnectionUtil getJdbcConnectionUtil(String jdbc
 	
 	@Setup
 	public void setup() throws SQLException, PropertyVetoException {
-		FhirContext fhirContext = FhirContext.forR4();
+		log.info("Starting setup for stage " + stageIdentifier);
+		fhirContext = FhirContext.forR4();
```

Comment:
Sorry I missed this comment in the previous round (thanks to GitHub review tool!). The reason we actually create and keep the `fhirContext` in `setup` is because it is expensive and we only do it once per worker. Your suggestion is also a good one. But, interestingly, the `Cached`  version was not available in the HAPI version we were using. So I increased the HAPI version in `pom.xml` and switched to cached version.

The stale dependency issue should go away once we fix #243 (soon).

Reply:
I spoke too soon: Increasing the version of HAPI has some R3->R4 implications too. Those seem to be easily fixable too but I don't want to expand the scope of this PR much more. For now, I added a TODO to fix this once we move to a newer HAPI.

Reply:
Resolving this conversation and submitting this PR as I have a huge formatting change PR which I prefer to apply on top of this (instead of dealing with formatting issues of this again). I think I have addressed all comments but please feel free to comment more and I will address in a separate PR.